day12/writex.py

```diff
@@ -0,0 +1,51 @@
+def writeX(matrix):
+    rows = set()
+    cols = set()
+
+    height = len(matrix[0])
+    width = len(matrix)
+
+    for i in xrange(height):
+        for j in xrange(width):
+            if matrix[i][j] == 'x':
+                rows.add(i)
+                cols.add(j)
+
+    for row in rows:
+        for j in xrange(width):
+            if matrix[row][j] != 'x':
+                matrix[row][j] = 'x'
+
+    for i in xrange(height):
```

Comment:
there's an inconsistency between this loop and the one before

    for row in rows:
        for j in xrange(width):

vs 

    for i in xrange(height):
        for col in cols: